odl/trafos/backends/pywt_bindings.py

```diff
@@ -0,0 +1,799 @@
+    if max_levels == 0:
+        raise ValueError('data size {} too small for transform, results '
+                         'in maximal `nlevels` of 0'.format(data_len))
+    if not 0 < nlevels <= max_levels:
+        raise ValueError('`nlevels` must satisfy 0 < nlevels <= {}, got {}'
+                         ''.format(max_levels, nlevels))
+
+
+def pywt_coeff_shapes(shape, wavelet, nlevels, mode):
+    """Return a list of coefficient shapes in the specified transform.
+
+    The wavelet transform specified by ``wavelet``, ``nlevels`` and
+    ``mode`` produces a sequence of approximation and detail
+    coefficients. This function computes the shape of those
+    coefficient arrays and returns them as a list.
+
+    Parameters
+    ----------
+    shape : sequence
+        Shape of an input to the transform.
+    wavelet : string or `pywt.Wavelet`
+        Specification of the wavelet to be used in the transform.
+        If a string is given, it is converted to a `pywt.Wavelet`.
+        Use `pywt.wavelist` to get a list of available wavelets.
+    nlevels : positive int
+        Number of scaling levels to be used in the decomposition. The
+        maximum number of levels can be calculated with
+        `pywt.dwt_max_level`.
+    mode : string, optional
+        PyWavelets style signal extension mode. See `signal extension modes`_
+        for available options.
+
+    Returns
+    -------
+    shapes : list
+        The shapes of the approximation and detail coefficients at
+        different scaling levels in the following order:
+
+            ``[shape_aN, shape_DN, ..., shape_D1]``
+
+        Here, ``shape_aN`` is the shape of the N-th level approximation
+        coefficient array, and ``shape_Di`` is the shape of all
+        i-th level detail coefficients.
+
+    See Also
+    --------
+    pywt.dwt_coeff_len : function used to determine coefficient sizes at
+        each scaling level
+
+    Examples
+    --------
+    Determine the coefficient shapes for 2 scaling levels in 3 dimensions
+    with zero-padding. Approximation coefficient shape comes first, then
+    the level-2 detail coefficient and last the level-1 detail coefficient:
+
+    >>> pywt_coeff_shapes(shape=(16, 17, 18), wavelet='db2', nlevels=2,
+    ...                   mode='zpd')
+    [(6, 6, 6), (6, 6, 6), (9, 10, 10)]
+
+    References
+    ----------
+    .. _signal extension modes:
+       https://pywavelets.readthedocs.io/en/latest/ref/signal-extension-\
+modes.html
+    """
+    shape = tuple(shape)
+    if any(int(s) != s for s in shape):
+        raise ValueError('`shape` may only contain integers, got {}'
+                         ''.format(shape))
+    wavelet = pywt_wavelet(wavelet)
+
+    nlevels, nlevels_in = int(nlevels), nlevels
+    if nlevels_in != nlevels:
+        raise ValueError('`nlevels` must be integer, got {}'
+                         ''.format(nlevels_in))
+    # TODO: adapt for axes
+    _check_nlevels(nlevels, min(shape), wavelet)
```

Comment:
This makes the whole function kind of superfluous. Especially since one needs `max_levels` outside to print the proper error message. Not so bad probably since it's used twice only.